lib/kramdown/converter/kramdown.rb

```diff
@@ -186,8 +186,9 @@ def convert_dt(el, opts)
 
       def convert_html_element(el, opts)
         markdown_attr = el.options[:category] == :block && el.children.any? do |c|
-          c.type != :html_element && (c.type != :p || !c.options[:transparent]) &&
-            Element.category(c) == :block
+          c.type != :html_element &&
+          !(c.type == :p && c.options[:transparent] && c.children.none? { |t| ![:entity, :text, :html_element].member?(t.type) }) &&
```

Comment:
`[:entity, :text, :html_element]` is a static array. In order to avoid repeated allocations of this array on multiple iterations, I recommend hoisting the array as a private constant:
```ruby
TYPES_ARRAY = [:entity, :text, :html_element]
private_constant :TYPES_ARRAY

def convert_html_element(el, opts)
  markdown_attr = el.options[:category] == :block && el.children.any? do |c|
    ...
    c.children.none? { |t| !TYPES_ARRAY.include?(t.type) }
    ...
  end
end
```

Reply:
How did you determine these three element types?

Reply:
@gettalong, I added element types until the rest of the Kramdown spec suite passed without changes. 😄 

* * *

I considered 

```ruby
c.children.none? { |t| MARKDOWN_TYPES.member?(t.type) }
```

instead of 

```ruby
c.children.none? { |t| !NON_MARKDOWN_TYPES.member?(t.type) }
```

but I think the list would be longer.